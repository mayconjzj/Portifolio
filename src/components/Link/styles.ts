import tw from 'tailwind-styled-components';

export const Link = tw.a`
  text-blue-500
`;
