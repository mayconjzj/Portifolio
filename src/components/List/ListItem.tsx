export const ListItem = ({ children }: { children: React.ReactNode }) => {
  return <li className="list-none">{children}</li>;
};
