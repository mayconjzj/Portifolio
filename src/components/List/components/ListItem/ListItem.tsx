export const ListItem = ({ children }: { children: React.ReactNode }) => {
  return <li>{children}</li>;
};
