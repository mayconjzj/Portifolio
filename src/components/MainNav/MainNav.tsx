export const MainNav = () => {
  return (
    <nav>
      <ul>
        <li>Home</li>
        <li>Sobre mim</li>
        <li>Projetos</li>
        <li>Contato</li>
      </ul>
    </nav>
  );
};
