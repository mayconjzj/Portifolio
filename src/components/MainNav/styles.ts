import tw from 'tailwind-styled-components';

export const MainNav = tw.nav`
  md:flex
  hidden
`;
