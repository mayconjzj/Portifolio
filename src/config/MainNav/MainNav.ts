import { MainNavProps } from '@/models';

export const MainNavConfig: MainNavProps[] = [
  {
    title: 'Home',
    href: '#home'
  },
  {
    title: 'Sobre mim',
    href: '#about'
  },
  {
    title: 'Projetos',
    href: '#projects'
  },
  {
    title: 'Contato',
    href: '#contact'
  }
];
