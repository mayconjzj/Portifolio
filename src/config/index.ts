export * from './HomePage';
export * from './MainNav';
export * from './Logo';
