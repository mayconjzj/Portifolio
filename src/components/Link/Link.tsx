import * as S from './styles';

export const Link = ({
  children,
  href,
  ...props
}: {
  children: React.ReactNode;
  href: string;
}) => {
  return (
    <S.Link href={href} {...props}>
      {children}
    </S.Link>
  );
};
