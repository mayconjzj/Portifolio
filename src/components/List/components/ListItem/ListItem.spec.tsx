import Link from 'next/link';

import { render } from '@testing-library/react';

import { MainNavConfig } from '@/config/MainNav';

import { ListItem } from './ListItem';

describe('Header', () => {
  it('should render the header', () => {
    const { asFragment } = render(
      <>
        {MainNavConfig.map((item) => (
          <ListItem key={item.title}>
            <Link href={item.href}>{item.title}</Link>
          </ListItem>
        ))}
      </>
    );
    expect(asFragment()).toBeTruthy();

    expect(asFragment()).toMatchSnapshot();
  });
});
