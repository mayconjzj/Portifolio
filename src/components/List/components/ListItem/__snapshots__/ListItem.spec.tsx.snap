// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the header 1`] = `
<DocumentFragment>
  <li>
    <a
      href="/"
    >
      Home
    </a>
  </li>
  <li>
    <a
      href="/about"
    >
      Sobre mim
    </a>
  </li>
  <li>
    <a
      href="/projects"
    >
      Projetos
    </a>
  </li>
  <li>
    <a
      href="/contact"
    >
      Contato
    </a>
  </li>
</DocumentFragment>
`;
